app/models/recipe.rb

```diff
@@ -1,3 +1,24 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def all_ingredients
+      #there may be a better way to do this with AR but considering the circumstances this will do for now
+      list = []
+      self.ingredients.each do |ingredient|
+         list << ingredient.name
+      end
```

Comment:
Your gut is right - you can do this with AR! And it's *easier* in my opinion! Check it out:
`ingredients.pluck(:name)` will return to you an array of names of the ingredients associated with this recipe.